### DISCUSSION TITLE
Idea: Suggest *static* method syntax in unit tests?

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [Codeception/Codeception#6043] [Ideas]
When I'm using the syntax recommended at https://codeception.com/docs/05-UnitTests#Unit-Testing in a unit test
```php
$this->assertSame(1,1);
```
...I'm getting this error from phpstan:
> Dynamic call to static method PHPUnit\Framework\Assert::assertSame().

So the (for phpstan) correct way would be:
```php
self::assertSame(1,1);
```

However, https://phpunit.readthedocs.io/en/9.3/assertions.html#static-vs-non-static-usage-of-assertion-methods says about `->` vs. `::`:

> The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

What do you think?

**Comment by Naktibalda** [comment]
I think that it would increase confusion if it was recommended to use static assertions in Unit tests but the same methods provided by Asserts module would have to be called non-statically.

Unit tests is the area where Codeception doesn't provide much over standard PHPUnit functionality, so they should be coded in PHPUnit way and it isn't up to us to set best practices for PHPUnit tests.
